src/main/java/com/icthh/xm/ms/entity/web/rest/FunctionResource.java

```diff
@@ -184,4 +184,16 @@ public static String getFunctionKey(HttpServletRequest request) {
         String bestMatchingPattern = request.getAttribute(BEST_MATCHING_PATTERN_ATTRIBUTE).toString();
         return new AntPathMatcher().extractPathWithinPattern(bestMatchingPattern, path);
     }
+
+    private ResponseEntity<Object> processCreatedResponse(FunctionContext result) {
+        return Optional.ofNullable(result.getId())
+            .map(id -> URI.create(ENTITY_NAME_FUNCTION_PATH + id))
+            .map(uri -> ResponseEntity.created(uri)
+                .headers(HeaderUtil.createEntityCreationAlert(ENTITY_NAME_FUNCTION_CONTEXT, String.valueOf(result.getId())))
+                .body(result.functionResult()))
+            .orElse(ResponseEntity
```

Comment:
propose to use orElseGet() to avoid redundant ResponseEntity creation

Reply:
@amedvedchuk In 99 percents where will be no id, so according to https://github.com/commonquail/java-optional-or-else-benchmark esscape analysis will not crteate any optional and RE will return answer, so there ae no need of anonymous lambda.

Reply:
@sergeysenja1992 'lazy' is not an argument. I edded reason why this implemented in such way.